### PR TITLE
Consolidate global search settings (AOD, Basic)

### PR DESCRIPTION
**Pull request by Dillon-Brown**
## Description

- Removed a duplicate "Global Search" link in the admin panel.
- Moved AOD to "Search Settings" page.
- Moved SearchModules to "Search Settings" page. 
- Removed AOD enable option, this now comes from the search engine drop-down.
- Minor coding standard fixes, code simplification, cleanup, etc.
- Minor styling fixes, each of the search settings should be using standard panels:
- Cleaned up flaky ElasticSearch tests (These should be improved further / overhauled if we upgrade ElasticSearch versions).

searchsettings2

## Motivation and Context

- This is to consolidate and simplify the search options within the CRM. All settings should work exactly the same as before just now in one location.

## How To Test This

1. Check AOD works as expected.
3. Check basic search works as expected.
4. Check module selection works as expected.
5. Check that all settings visible on the page work and that nothing is missing.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [X] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [X] My change requires a change to the documentation.
- [X] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

File: modules/Administration/Search/MVC/View.php [hunk 1]
```diff
@@ -77,8 +78,12 @@ public function preDisplay()
         if (!isset($sugar_config['search']) || !$sugar_config['search']) {
             LoggerManager::getLogger()->warn('Configuration does not contains default search settings.');
         }
-        $search = isset($sugar_config['search']) ? $sugar_config['search'] : null;
+        $search = $sugar_config['search'] ?? null;
```
Comment (clemente-raposo) [inline modules/Administration/Search/MVC/View.php:81]: Hi @Dillon-Brown can we use `??` in this branch? do we still have to keep `5.6` support?

Comment (Dillon-Brown) [reply modules/Administration/Search/MVC/View.php:81]: Develop will be PHP 7.3 minimum (Currently 7.2 until the composer update PR get merged).

File: modules/Administration/Search/MVC/View.php [hunk 2]
```diff
@@ -77,8 +78,12 @@ public function preDisplay()
...
         $this->smarty->assign('config', $search);
+
+        $cfg = new Configurator();
+        $this->smarty->assign('aod_config', $cfg->config['aod']);
```
Comment (clemente-raposo) [inline modules/Administration/Search/MVC/View.php:85]: Hi @Dillon-Brown, where `aod_config` is used in the html/tpl?

Comment (Dillon-Brown) [reply modules/Administration/Search/MVC/View.php]: Unused, nice spot. Pushed!

**Review by jack7anderson7 — APPROVED** [review]

File: modules/Administration/Search/MVC/View.php (hunk 2 above)
Comment (clemente-raposo) [reply modules/Administration/Search/MVC/View.php]: thanks

**Review by clemente-raposo — DISMISSED** [review]
code looks ok

**Review by clemente-raposo — DISMISSED** [review]
latest changes look ok

**Review by clemente-raposo — APPROVED** [review]
latest changes look ok

**Review by Mac-Rae — APPROVED** [review]

**Comment by Dillon-Brown** [comment]
Latest push is just rebasing onto develop, no real changes.

**Review by clemente-raposo — APPROVED** [review]
latest changes look ok

**Review by Mac-Rae — APPROVED** [review]

**Pull request merged**